src/common/components/FilterMenu.tsx

```diff
@@ -0,0 +1,87 @@
+import { Box, useColorModeValue } from '@chakra-ui/react';
+import { useCallback, useState } from 'react';
+import { IconType } from 'react-icons';
+import { BsChevronDown } from 'react-icons/bs';
+
+import { Button } from '../../ui/Button';
+import { Icon } from '../../ui/Icon';
+import { Menu } from '../../ui/Menu';
+import { MenuButton } from '../../ui/MenuButton';
+import { MenuItem } from '../../ui/MenuItem';
+import { MenuList } from '../../ui/MenuList';
+
+interface MenuItem {
+  onClick: () => void;
+  label: string;
+}
+
+interface FilterMenuProps {
+  filterLabel: string | (() => string);
+  menuItems: MenuItem[];
+  leftIcon: IconType;
+}
+
+export function FilterMenu({ filterLabel, menuItems, leftIcon }: FilterMenuProps) {
+  const borderColor = useColorModeValue('slate.300', 'slate.900');
+  const [isHovered, setIsHovered] = useState(false);
+  const handleMouseEnter = useCallback(() => {
```

Comment:
Using useCallback in these scenarios could lead to worse performance, as the cost of memoization outweighs the benefits for very simple functions.